test/e2e_env/universal/meshretry/http.go

```diff
@@ -71,26 +87,114 @@ spec:
 				universal.Cluster, "demo-client", "test-server.mesh",
 			)
 			g.Expect(err).ToNot(HaveOccurred())
-		}).Should(Succeed())
-		Consistently(func(g Gomega) {
-			// --max-time 8 to wait for 8 seconds to beat the default 5s connect timeout
+		}, "30s", "1s", MustPassRepeatedly(5)).Should(Succeed())
+
+		By("Adding a MeshFaultInjection for test-server")
+		Expect(universal.Cluster.Install(YamlUniversal(meshFaultInjection))).To(Succeed())
+
+		By("Check some errors happen")
+		var errs []error
+		for i := 0; i < 50; i++ {
+			time.Sleep(time.Millisecond * 100)
+			_, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client", "test-server.mesh",
+			)
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
+		Expect(errs).ToNot(BeEmpty())
```

Comment:
can't we express the same thing but using `Eventually`?

Reply:
also, you should probably `break` the loop when there is at least 1 error in `errs`, but I'd go with `Eventually`